### PR TITLE
[CARBONDATA-2856][BloomDataMap] Fix bug in bloom index on multiple dictionary columns

### PR DESCRIPTION
**Pull request by xuchuanyin**
During direct generating bloom index for dictionary and date columns, we
will use MdkKeyGenerator to get the surrogate key. But the
MdkKeyGenerator is for all MDK, so we need to pad fake bytes to mdk.
Finally, the generator will only deal with the real parts of the mdk and
get the final value.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 `NO`
 - [x] Any backward compatibility impacted?
 `NO`
 - [x] Document update required?
`NO`
 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
`Test modified to cover this problem`
        - How it is tested? Please attach test report.
`Tested in local machine`
        - Is it a performance related change? Please attach the performance test report.
`NO`
        - Any additional information to help reviewers in testing this change.
      `NA` 
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
`NA`

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7913/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6637/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7914/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6638/

**Comment by jackylk** [comment]
retest sdv please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6300/

**Comment by xuchuanyin** [comment]
retest sdv please

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/8029/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6752/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6782/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/8059/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/18/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/8076/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/8084/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/24/

**Comment by jackylk** [comment]
LGTM

File: datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomDataMapWriter.java [hunk 1]
```diff
@@ -91,30 +91,28 @@
...
+      totalSize += size;
+    }
+    byte[] fakeMdkBytes = new byte[totalSize];
+
+    // put this bytes to corresponding position
+    int thisKeyIdx = indexCol2MdkIdx.get(indexColumns.get(indexColIdx).getColName());
+    int destPos = 0;
+    for (int keyIdx = 0; keyIdx < columnarSplitter.getBlockKeySize().length; keyIdx++) {
+      if (thisKeyIdx == keyIdx) {
+        System.arraycopy(value, 0, fakeMdkBytes, destPos,
```
Comment (jackylk) [inline datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomDataMapWriter.java]: I am not quite sure about this, please @ravipesala have a look

Comment (ravipesala) [reply datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomDataMapWriter.java]: Please don't  copy, directly convert to int using ` CarbonUtil.getSurrogateInternal(data, startOffsetOfData, columnValueSize)`

Comment (xuchuanyin) [reply datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomDataMapWriter.java]: nice, it works. :+1:

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6474/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/8163/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/92/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**